### ISSUE TITLE
Rebuild data model, test functionality with schematic 24.1.1

### ISSUE THREAD

**Issue by Bankso** [mc2-center/data-models#61]
Schematic v24.1.1 includes a refactor to how data models are converted to JSON-LD - following its release (planned for 1/22/24), FAIR is requesting that [DCCs test the functionality of the refactor by Feb 1st](https://docs.google.com/presentation/d/13GnpCKe_BFVfMWiSOs86ZQfd2_oQgC9c7Do2OfDZ3Xk/edit#slide=id.g265fb022bdc_0_5).

Refactor details and guidelines for testing are outlined here: https://sagebionetworks.jira.com/wiki/spaces/SCHEM/pages/3062530050/Schematic+Schema+Refactor#Testing

Note that the request includes testing the CLI and DCA (DCA staging will be updated to schematic 24.1.1 by Jan 26th; production by Feb 1st)

Suggested data-model repo actions:
- create a branch in data-models for testing the new schematic version, based on the current content of main
- create a new virtual environment and install schematicpy==24.1.1
- update testing branch with rebuilt model

Testing actions:
- CLI tests can be carried out by pointing the schematic config.yml to the rebuilt model
- DCA testing in staging will require an update to the MC2 config in [Sage-Bionetworks/data_curator_config/staging](https://github.com/Sage-Bionetworks/data_curator_config/tree/staging)
- Production DCA will require an update to the MC2 config in [Sage-Bionetworks/data_curator_config/prod](https://github.com/Sage-Bionetworks/data_curator_config/tree/prod)

**Comment by Bankso** [comment]
Branch in data-models associated with this issue: https://github.com/mc2-center/data-models/tree/schematic-refactor-issue-61

**Comment by aclayton555** [comment]
Orion will incorporate some values into data model, push a release, then complete this testing

**Comment by Bankso** [comment]
Current plan:

- [x] complete model updates in https://github.com/mc2-center/data-models/tree/curation-terms-1-24
    - [x] edits from publication review
    - [x] edits from dataset review
    - [x] edits from tool review 
- [x] merge into https://github.com/mc2-center/data-models/tree/schematic-refactor-issue-61
- [x] update valid values, rebuild CSV, JSON-LD
- [x] regenerate templates from new model
- [x] Release new model version
- [x] Update DCA config
- [ ] ~make PR to DCA staging branch~
- [x] test functionality of templates (including validation)
- [x] make PR to DCA prod branch, once data model and template functionality is verified

**Comment by Bankso** [comment]
Data model updates are ready for review in PR #70. Subsequent testing and DCA updates will be logged as part of this issue, as well.

**Comment by Bankso** [comment]
Data model and template testing with schematic 24.1.1 (Note that DCA testing will be logged separately)

- [x] data model CSV can be read by schematic
- [x] data model can be converted to JSON-LD
- [x] manifest templates can be created from newly generated model (xlsx and sheets)
- [x] manifest templates include all expected columns, descriptions, notes, formatting, valid values
   - [x] PublicationView
   - [x] DatasetView
   - [x] ToolView
   - [x] EducationalResource
   - [x] Biospecimen
   - [x] GeoMx Level 1
   - [x] GeoMx Level 2
   - [x] GeoMx Level 3
   - [x] GeoMx ROI/Annotations
   - [x] GeoMx Imaging
- [x] manifest templates can be validated by schematic
   - [x] PublicationView
   - [x] DatasetView
   - [x] ToolView
   - [x] EducationalResource
   - [x] Biospecimen
   - [x] GeoMx Level 1
   - [x] GeoMx Level 2
   - [x] GeoMx Level 3
   - [x] GeoMx ROI/Annotations
   - [x] GeoMx Imaging
- [ ] ~manifest templates can be submitted to Synapse by schematic~
- I'll test this through the DCA and go back to schematic CLI if any issues pop up

**Comment by Bankso** [comment]
Testing updated DCA (MC2 data model v5.1.1, DCA v24.2.1, schematic v24.1.1)
Manifest generation, validation, and submission for templates:
   - [ ] PublicationView
   - [ ] DatasetView
   - [ ] ToolView
   - [ ] EducationalResource
   - [ ] Biospecimen
   - [ ] GeoMx Level 1
   - [ ] GeoMx Level 2
   - [ ] GeoMx Level 3
   - [ ] GeoMx ROI/Annotations
   - [ ] GeoMx Imaging

**Comment by Bankso** [comment]
Currently blocked on this issue wrt to DCA testing, since I'm regularly getting time out errors. Schematic appears to play nice with all manifests so far, so I don't expect to see any problems once the API timeout issues are resolved.

**Comment by Bankso** [comment]
DCA updated to use schematic 24.2.1 as of 3/1/24. Manifest generation times are still long, but I was able to generate an Educational Resource manifest that was correctly populated with metadata from Synapse. I will test for each manifest type, but I imagine this will be consistent across components.

Evaluating manifest validation and submission:
   - [x] PublicationView
   - [x] DatasetView: lists of URLs are causing errors. This is seen with schematic, too. I think we should change the validation rule to "URL::list like" or something similar to address the validation errors.
   - [x] ToolView
   - [x] EducationalResource
   - [x] Biospecimen
   - [x] GeoMx Level 1
   - [x] GeoMx Level 2
   - [x] GeoMx Level 3
   - [x] GeoMx ROI/Annotations
   - [x] GeoMx Imaging

**Comment by aclayton555** [comment]
With recent DCA fix, Orion will continue testing validation and submission. Should be able to close this out at the start of the next sprint. Testing may inform some validation rules (e.g. URLs separated by comma)

**Comment by aclayton555** [comment]
Confirming this work will roll into 24-3 sprint.
